src/ember-app.js

```diff
@@ -267,11 +219,22 @@ class EmberApp {
    *
    * @returns {Promise<Ember.ApplicationInstance>} instance
    */
-  buildAppInstance() {
-    return this.app.boot().then(function(app) {
-      debug('building instance');
-      return app.buildInstance();
-    });
+  async buildAppInstance() {
+    let sandbox = this.buildSandbox();
+
+    debug('adding files to sandbox');
+
+    for (let script of this.scripts) {
+      debug('evaluating file %s', script);
+      sandbox.runScript(script);
+    }
+
+    debug('files evaluated');
+
+    let app = await this.createEmberApp(sandbox).boot();
```

Comment:
I think you may want to `this.app` instead of a local variable here.

Reply:
I was trying to avoid _ever_ setting `this.app`, specifically to ensure that we don't "leak" it or accidentally allow it to live longer than it should.

I'll definitely fix up the destroy code though (which you called out in another comment above).